src/components/Clusters/ClusterDetail.tsx

```diff
@@ -1,5 +1,140 @@
+import { Button, Card, CardSection, Group, Text } from '@mantine/core';
+import { useState } from 'react';
+import { Helmet } from 'react-helmet-async';
+import { RiDeleteBin5Line, RiSettings4Line, RiToggleFill, RiToggleLine } from 'react-icons/ri';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import { useAuth } from '@/hooks/useAuth';
+import { useClusterDetails, useJobsOnCluster } from '@/hooks/useClusters';
+import { disableModal, enableModal } from '@/utils/modals';
+import { pluralize } from '@/utils/string';
+
+import ClusterConfigCard from './ClusterConfigCard';
+import ClusterJobsCard from './ClusterJobsCard';
+import ClusterLabelBadge from './ClusterLabelBadge';
+import ClusterOwnerDetails from './ClusterOwnerDetails';
+
 function ClusterDetail() {
-  return <div className="flex h-full flex-col" />;
+  const navigate = useNavigate();
+  const { clusterId: clusterName = '' } = useParams();
+  if (!clusterName) navigate('/404', { replace: true });
```

Comment:
Thinking aloud I don't think this check is necessary. This component should only be mapped to a route where clusterName will always exist?

Reply:
Instead, I think you should navigate to 404 after fetching the cluster.

const { data: cluster = null } = useClusterDetails(clusterName);
if (!cluster) navigate('/404', { replace: true });

So always try to fetch a cluster. If that does not that exist then navigate to 404.